modules/exploits/linux/http/panos_auth_rce.rb

```diff
@@ -0,0 +1,140 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Networks Authenticated Remote Code Execution',
+        'Description' => %q{
+          An OS Command Injection vulnerability in the PAN-OS management interface that allows authenticated administrators to execute arbitrary OS commands with root privileges.
+
+          This issue impacts:
+
+          PAN-OS 9.0 versions earlier than 9.0.10;
+
+          PAN-OS 9.1 versions earlier than 9.1.4;
+
+          PAN-OS 10.0 versions earlier than 10.0.1.
+        },
+        'Author' => [
+          'Mikhail Klyuchnikov and Nikita Abramov', # Vulnerability discovery
+          'UnD3sc0n0c1d0',                          # Exploit
+          'jheysel-r7'                              # msf module
+        ],
+        'References' => [
+          ['CVE', '2020-2038'],
+          ['URL', 'https://swarm.ptsecurity.com/swarm-of-palo-alto-pan-os-vulnerabilities/'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2020-2038']
+        ],
+        'DisclosureDate' => '2020-09-09',
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Arch' => ARCH_CMD,
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux x86',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X86,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ],
+          [
+            'Linux x86_64',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X64,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => { 'WfsDelay' => 2 },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptString.new('USERNAME', [false, 'PAN-OS username', 'admin']),
+        OptString.new('PASSWORD', [false, 'Password for username', 'admin'])
```

Comment:
Any username, or admin specifically?

Reply:
Administrators specifically, thanks for catching this!